### PR TITLE
0% AB test removing the sticky nav

### PR DESCRIPTION
**Pull request by holder-josh**
This is an initial 0% AB test that replaces the current sticky.js code with a much simplified version that only sticks the top ad.

The current sticky nav implementation is a blocker on the 'toast' element of the upcoming liveblog work (which has a sticky top button) and is also the cause of all of these Github issues: https://github.com/guardian/frontend/labels/sticky-nav

Putting this out there as not prod-ready code but something we can show to people and get feedback on.

Final implementation could be simpler as some of the injected CSS here could be done in the CSS files. 

May need to add some extra workarounds for the Apple banner/adblock banner. 

@OliverJAsh, you may be interested in this?

nav

### CODE REVIEW

**Comment by rich-nguyen** [comment]
I haven't even checked the code. :+1:

**Comment by ScottPainterGNM** [comment]
screen shot 2016-01-07 at 16 28 00

File: static/src/javascripts/projects/common/modules/navigation/newSticky.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                    this.$header.css({
+                        'margin-top': this.adHeight + 'px'
+                    });
+                }.bind(this));
+            }.bind(this));
+        }
+    };
+
+    return new StickyHeader();
+});
```
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: As noted in person, one of the things I didn't like about old sticky nav was the OOP. I think we could make this less buggy and easier to read without the class and object mutation. I think we generally prefer that style for the rest of our code (more so in Scala).

Not important for the test, though, so don't let this stop you! :smile:

**Comment by paperboyo** [comment]
sticky_no_more_lossy100

**Comment by crifmulholland** [comment]
:+1:  for the comments

File: static/src/javascripts/projects/common/modules/navigation/newSticky.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    'lodash/functions/bindAll',
+    'common/utils/detect'
+], function (
+    fastdom,
+    $,
+    mediator,
+    bindAll,
+    detect) {
+
+    function StickyHeader() {
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: Why not just have the init method's content in the constructor?

Then you could just define your module as

    function StickyHeader() {
         setup();
    }

    return new StickyHeader();

This way you wouldn't have to know if newSticky had already been initialized when you grab the service - you wouldn't be splitting definition and generation. In your test code you could just do

     let stickyHeader;
     beforeEach(()=> {
         stickyHeader = new stickyHeaderService.constructor();
     });

**Comment by Jholder112233** [comment]
Closing this as it's been replaced with https://github.com/guardian/frontend/pull/11503

Would be good to get comments on that code.

**Pull request closed without merging**